### PR TITLE
python3.7, dependency add, subnet read fix

### PR DESCRIPTION
**Pull request by cmdpwnd**
1. stop breaking built python; manually setup with `update-alternatives`
2. add support for python3.7
3. require `setuptools`
4. fix `index out of range` when reading subnet

### CODE REVIEW

**Review by slowr — APPROVED** [review]
~~Looks good to me and tested on Docker. Approved~~

File: lib/traixroute/handler/handle_pch.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python
```
Comment (slowr) [inline lib/traixroute/handler/handle_pch.py]: Changing to just python has the risk that the user does not have python3 installed at all.

File: setup/install.sh [hunk 2]
```diff
@@ -25,11 +25,10 @@ cur_dir=$(dirname $0)
 
 additional(){
 
-    sudo pip3 install --upgrade pip
-    sudo pip3 install --upgrade -r $cur_dir/requirements.txt
-    
+    sudo pip install --upgrade -r $cur_dir/requirements.txt
```
Comment (slowr) [inline setup/install.sh]: pip != pip3
we need to be sure that the tool installs the python3 dependencies and not python2

File: setup/install.sh [hunk 3]
```diff
@@ -25,11 +25,10 @@ cur_dir=$(dirname $0)
...
+
     if ! hash scamper 2>/dev/null; then
-        sudo python3 $cur_dir/../lib/traixroute/downloader/install_scamper.py
+        sudo python $cur_dir/../lib/traixroute/downloader/install_scamper.py
```
Comment (slowr) [inline setup/install.sh]: same as above

File: setup/install.sh [hunk 4]
```diff
@@ -45,14 +44,14 @@ if [ $OS = 'Darwin' ]; then
         echo 'Exit.'
     elif [ $installed -eq 0 ]; then 
         echo 'Xcode Command Line Tools are installed.'
-        python3 -V &>/dev/null;
+        python -V &>/dev/null;
```
Comment (slowr) [inline setup/install.sh]: same as above

File: setup/install.sh [hunk 5]
```diff
@@ -45,14 +44,14 @@ if [ $OS = 'Darwin' ]; then
...
         if [ $? -ne 0 ]; then
             echo 'Python 3 is not installed.'
             installer -pkg $cur_dir/python-3.4.4-macosx10.6.pkg -target /
             echo 'Installing dependencies for OS X.'
             additional
         else
-            python3 -c "import sys; exit(float(sys.version[0:3])>=3.4)"
+            python -c "import sys; exit(float(sys.version[0:3])>=3.4)"
```
Comment (slowr) [inline setup/install.sh]: same as above

**Review by slowr — CHANGES_REQUESTED** [review]
We cannot use just `python` because there is a possibility that the default alias directs to python2. Please change back the calls to python to `python3`

File: setup/install.sh (hunk 5 above)
Comment (cmdpwnd) [reply setup/install.sh]: pushed

File: setup/install.sh (hunk 4 above)
Comment (cmdpwnd) [reply setup/install.sh]: pushed

File: setup/install.sh (hunk 3 above)
Comment (cmdpwnd) [reply setup/install.sh]: pushed

File: setup/install.sh (hunk 2 above)
Comment (cmdpwnd) [reply setup/install.sh]: pushed

File: lib/traixroute/handler/handle_pch.py (hunk 1 above)
Comment (cmdpwnd) [reply lib/traixroute/handler/handle_pch.py]: pushed

**Comment by cmdpwnd** [comment]
Restored python3 commands, but make sure to setup your versions with `update-alternatives` if running multiple versions.

```
Eg:
sudo update-alternatives --install /usr/bin/pip3 pip3 /usr/local/bin/pip3.X 1
sudo update-alternatives --install /usr/bin/python3 python3 /usr/local/bin/python3.X 1
```

**Review by slowr — APPROVED** [review]

**Comment by nrodday** [comment]
The index-out-of-range fix is probably wrong. It solves the problem but compromises functionality as there are simply 0 subnet exported afterwards .

The format is: 
exchange_point_id, short_name, status, version, multicast, mlpa, subnet, participants
228,SOL-IX v4,Active,IPv4,Unknown,Available,193.110.12.0/24,69

Therefore the 7th position is correct for the subnet extraction. I would propose to increase the boundary for the check by 1 instead:

if len(temp_string) > 6:

**Pull request merged**